src/core/public/chrome/ui/header/collapsible_nav.tsx

```diff
@@ -0,0 +1,270 @@
+/*
```

Comment:
Writing some jest tests now but ultimately I don't think the tests I add will add much security over the overwhelming amount of tests that fail when the nav doesn't work right.

Because so much of the test suite already relies on the nav, it has pretty good coverage passively. 

I'm not saying that's perfect or anything, I'm literally running new tests locally right now, but I do think that tests shouldn't be a blocker for this PR.

Reply:
Tests added: https://github.com/myasonik/kibana/blob/new-nav/src/core/public/chrome/ui/header/collapsible_nav.test.tsx